src/edgepi/adc/edgepi_adc.py

```diff
@@ -980,6 +975,127 @@ def set_config(
                                         [None])
         self.__config(**args)
 
+    def adc1_config_and_read_samples_batch(
```

Comment:
This function is massive because it is a combination of set_config and read_sample, but also most of the child functions. One issue with this codebase is that `get_state` is called a large number of times, even when only a few properties are needed. The construction of the ADCState object, and sometimes reading from the hardware is quite slow. By combining all the functions, we can read the register state from the hardware just once, then reuse those values & update our local copy after we complete all the hardware writes.